executor/join.go

```diff
@@ -504,10 +504,10 @@ func (e *HashJoinExec) buildHashTableForList(innerResultCh <-chan *chunk.Chunk)
 		allTypes:  allTypes,
 		keyColIdx: innerKeyColIdx,
 	}
-	initList := chunk.NewList(allTypes, e.initCap, e.maxChunkSize)
-	e.rowContainer = newHashRowContainer(e.ctx, int(e.innerEstCount), hCtx, initList)
+	e.rowContainer = newHashRowContainer(e.ctx, int(e.innerEstCount), hCtx)
 	e.rowContainer.GetMemTracker().AttachTo(e.memTracker)
 	e.rowContainer.GetMemTracker().SetLabel(innerResultLabel)
+	e.rowContainer.SetSpillToDisk(e.ctx.GetSessionVars().MemQuotaHashJoin)
```

Comment:
We may still need the option for killing queries that exceeds memory quota.

Reply:
In #12116, I've separated different actions into each `memory.Tracker` , hence the action of `hashJoin.memTracker` takes control of spilling while the action of `e.ctx.GetSessionVars().Stmt` takes control of OOM killing and logging.